kim/mapper.py

```diff
@@ -90,6 +100,17 @@ def _extract_roles(self, base):
         _roles.update(getattr(cls, 'roles', None) or {})
         _roles.update(getattr(base, '__roles__', None) or {})
 
+        # Roles may be passed as list, convert to whitelist objects in this
+        # case
```

Comment:
I thought this was required but it actually wasn't in the end, but I decided to leave it in because it seems a bit weird having a mix of lists and Role objects and relying on them both supporting `__contains__` for it to work seamlessly.

Do you agree?